### PR TITLE
fixup cancellation

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs [hunk 1]
```diff
@@ -185,20 +184,20 @@ async Task Consumer_Received(object sender, BasicDeliverEventArgs eventArgs)
...
+                consumer.Model.BasicRejectAndRequeueIfOpen(eventArgs.DeliveryTag);
             }
             finally
             {
                 Interlocked.Decrement(ref numberOfMessagesBeingProcessed);
             }
         }
 
-        async Task Process(BasicDeliverEventArgs message)
+        async Task Process(BasicDeliverEventArgs message, CancellationToken messageProcessingCancellationToken)
```
Comment (adamralph) [inline src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs:200]: Allows CA2016 to do its work.

File: src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePumpConnectionFailedCircuitBreaker.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ void CircuitBreakerTriggered(object state)
         string name;
         TimeSpan timeToWaitBeforeTriggering;
         Timer timer;
-        Action<Exception> criticalErrorAction;
+        Action<string, Exception> criticalErrorAction;
```
Comment (adamralph) [inline src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePumpConnectionFailedCircuitBreaker.cs:62]: This class doesn't need to change. These changes revert it back to how it was before https://github.com/Particular/NServiceBus.RabbitMQ/pull/763

File: src/NServiceBus.Transport.RabbitMQ/Receiving/ModelExtensions.cs [hunk 3]
```diff
@@ -1,46 +1,25 @@
 ﻿namespace NServiceBus.Transport.RabbitMQ
 {
-    using System.Threading.Tasks;
```
Comment (adamralph) [inline src/NServiceBus.Transport.RabbitMQ/Receiving/ModelExtensions.cs:3]: The tasks used in this class are redundant.

At one time, the methods accepted a `TaskScheduler`, but [when that parameter was removed]((https://github.com/Particular/NServiceBus.RabbitMQ/commit/29080f305e7cf77c119da0c50e87060fb04a8db5#diff-963cdc5b2f2a1f7238c343f73499f46db12b5ba32483103a6e95dc50ce3c76e0)) in favour of `TaskScheduler.Default`, they became redundant. I mentioned this to @bording and he concurs.

(The use of `CancellationToken.None` in `TaskEx` drew my attention to this.)

**Review by DavidBoike — APPROVED** [review]

**Comment by DavidBoike** [comment]
@andreasohlund the Task stuff at the end baffles me and is over my head, but other than that I'm :+1:. Will leave to you to merge into the main PR tomorrow.

**Comment by adamralph** [comment]
Reviewed with @andreasohlund

**Pull request merged**